### ISSUE TITLE
Support Import Pre-process function

### ISSUE THREAD

**Issue by micahgodbolt** [salesforce-ux/theo#64]
This might take a bit of work, but thought I'd suggest it at least.

```YAML
props:
  myComponent:
    button:
      foreground:
        base: "#333"
        hover: "#000"
        active: "#333"
        disabled: "#777"
      background:
        base: "#fff"
        hover: "#ddd"
        active: "#fff"
        disabled: "#ccc"
```

output of this would be the same as if I strung all of them together (or delimiter of choice).
        
```YAML
  myComponent_button_foreground: "#333"
  myComponent_button_foreground_hover: "#000"
  ...
```

The gist here would be that if a prop is an object, that object is passed into the parsing function along with the parent string. And if that next value is an object, it's passed down again with the parent appended to the parent string. 

Turtles all the way down.

**Comment by micahgodbolt** [comment]
I've got a branch working to support nested props, but there is a desire to not include the code directly into this project. Instead I'd like to support some function you can expose in the Gulp task so that you can preprocess the data, hence supporting converting nested data to the flat data used by Theo.

**Comment by micahgodbolt** [comment]
fixed in #71
